include/permanent.hpp

```diff
@@ -322,34 +322,253 @@ std::complex<double> permanent_quad(std::vector<std::complex<double>> &mat) {
  * @param mat vector representing the flattened matrix
  * @return the permanent
  */
-double permanent_quad(std::vector<double> &mat) {
+double permanent_quad(std::vector<double> &mat) 
+{
     std::vector<qp> matq(mat.begin(), mat.end());
     qp perm = permanent(matq);
     return static_cast<double>(perm);
 }
 
-
-
 /**
  * \rst
  *
- * Returns the permanent of a matrix using Ryser's algo with Gray code ordering with fsum
+ * Returns the permanent of a matrix using Ryser's algo 
+ * with Gray code ordering with fsum
  *
  * \endrst
  *
  *
- * This is a wrapper around the templated function `libwalrus::perm_fsum` for Python
- * integration. It accepts and returns double numeric types, and
- * returns sensible values for empty and non-even matrices.
+ * This is a wrapper around the templated function `libwalrus::perm_fsum` 
+ * for Python integration. It accepts and returns double numeric types, 
+ * and returns sensible values for empty and non-even matrices.
  *
  *
  * @param mat vector representing the flattened matrix
  * @return the permanent
  */
-double permanent_fsum(std::vector<double> &mat) {
+double permanent_fsum(std::vector<double> &mat) 
+{
     std::vector<double> matq(mat.begin(), mat.end());
     double perm = perm_fsum(matq);
     return static_cast<double>(perm);
 }
 
+
+/**
+ * Returns the permanent of a matrix (nthreads=1)
+ *
+ * \rst
+ *
+ * Returns the permanent of a matrix using the BBFG algorithm.
+ * This algorithm was given by Glynn (2010) with the time-complexity 
+ * of \f$O(n 2^n)\f$ using Gray code ordering.
+ *
+ * \endrst
+ *
+ *
+ * @param mat  a flattened vector of size \f$n^2\f$, representing an
+ *      \f$n\times n\f$ row-ordered symmetric matrix.
+ * @return permanent of the input matrix
```

Comment:
It would be a good idea to give a `@tparam`  mentioning the expected types of `T`

Reply:
Not sure what to write there

Reply:
Something like "Type of the matrix data" should suffice.